packages/utils/src/crypto/packet/filler.ts

```diff
@@ -0,0 +1,47 @@
+import { u8aXOR } from '../../u8a'
+import { PRG } from '../prg'
+import { derivePRGParameters } from './keyDerivation'
+
+/**
+ * Writes the filler bitstring into the header such
+ * that the integrity tag can be computed
+ * @param header header u8a to write the filler
```

Comment:
why not return a header rather than mutating an array?

Reply:
The idea is that the header generation algorithm closely follows the description in the paper which means adding the components one-by-one to the header and then shifting the header to the end.

Reply:
not really? You can closely follow the paper without mutating an array the whole way...